sdks/java/harness/src/main/java/org/apache/beam/fn/harness/FnApiDoFnRunner.java

```diff
@@ -460,14 +461,22 @@ public void accept(WindowedValue input) throws Exception {
       // Extract out relevant TimerFamilySpec information in preparation for execution.
       for (Map.Entry<String, TimerFamilySpec> entry :
           parDoPayload.getTimerFamilySpecsMap().entrySet()) {
-        String timerFamilyId = entry.getKey();
-        TimeDomain timeDomain =
-            DoFnSignatures.getTimerSpecOrThrow(
-                    doFnSignature.timerDeclarations().get(timerFamilyId), doFn)
-                .getTimeDomain();
+        String timerIdOrTimerFamilyId = entry.getKey();
+        TimeDomain timeDomain;
+        if (timerIdOrTimerFamilyId.startsWith(TimerFamilyDeclaration.PREFIX)) {
+          timeDomain =
+              DoFnSignatures.getTimerFamilySpecOrThrow(
+                      doFnSignature.timerFamilyDeclarations().get(timerIdOrTimerFamilyId), doFn)
+                  .getTimeDomain();
+        } else {
+          timeDomain =
+              DoFnSignatures.getTimerSpecOrThrow(
+                      doFnSignature.timerDeclarations().get(timerIdOrTimerFamilyId), doFn)
+                  .getTimeDomain();
+        }
         Coder<Timer<Object>> timerCoder =
             (Coder) rehydratedComponents.getCoder(entry.getValue().getTimerFamilyCoderId());
-        timerFamilyInfosBuilder.put(timerFamilyId, KV.of(timeDomain, timerCoder));
+        timerFamilyInfosBuilder.put(timerIdOrTimerFamilyId, KV.of(timeDomain, timerCoder));
```

Comment:
Could you please add more javadoc/comments about why it's `timerId` or `timerFamilyId`?

Reply:
done.